emcc.py

```diff
@@ -1330,6 +1330,7 @@ def add_js_deps(sym):
       for sym in settings.EXPORTED_RUNTIME_METHODS:
         add_js_deps(shared.demangle_c_symbol_name(sym))
     if settings.ASYNCIFY:
+      settings.ASYNCIFY_IMPORTS_EXCEPT_JS_LIBS = settings.ASYNCIFY_IMPORTS[:]
       settings.ASYNCIFY_IMPORTS += ['*.' + x for x in js_info['asyncFuncs']]
```

Comment:
What if we instead just remove this line?  (i.e. don't include asyncFuncs in ASYNCIFY_IMPORTS at all)?   

I don't see ASYNCIFY_IMPORTS used anywhere else so just slightly changing its meaning seems fine.

Reply:
I think they are still passed to Wasm-opt for Asyncify preprocessing of the Wasm module, and they still need to be processed.

Reply:
Yup, used here for Asyncify

https://github.com/emscripten-core/emscripten/blob/453e83a3a1b8b9836c136400a2fd2e981a033a4a/emcc.py#L672

and here for JSPI

https://github.com/emscripten-core/emscripten/blob/453e83a3a1b8b9836c136400a2fd2e981a033a4a/emcc.py#L696

Reply:
OK fair enough.  I still don't love this new setting name though.

How about keeping `ASYNCIFY_IMPORTS` as the setting specified on the command line and creating a new settings which is the sub of both which we can pass to wasm-opt?

Reply:
> I still don't love this new setting name though.

I'm open to better names. I mean, it's purely an internal setting, not exposed to the user, so shouldn't matter too much other than for readability / clarity of what it is.

> and creating a new settings which is the sub of both which we can pass to wasm-opt?

Hm sorry I'm not sure I understand the suggestion. Do you mean we'd instead change all places where `ASYNCIFY_IMPORTS` is currently used to something like `ASYNCIFY_IMPORTS_AND_JS_FUNCS`?

That could work too but seemed like more work than changing the one / only place where we don't want JS funcs.

Reply:
If you do want to stash/copy the original list.. how about `USER_ASYNCIFY_IMPORTS`.. to match the existing `USER_EXPORTED_FUNCTIONS`

Reply:
I see.  I'm happy to land this change either way.

If we want to keep pulling the thread though.. can we get rid of `DEFAULT_ASYNCIFY_IMPORTS`?  Currently its just `DEFAULT_ASYNCIFY_IMPORTS = ['__asyncjs__*']`.   Can we instead tag all the EM_JS function that start with `__asyncjs__` as `.isAsync = true` rather then rely on this regex?  Then they would be more like JS library funcs in that respect and we would no longer need `DEFAULT_ASYNCIFY_IMPORTS` at all.. and `USER_ASYNCIFY_IMPORTS` would then also be meaningful?

Reply:
> If we want to keep pulling the thread though.. can we get rid of `DEFAULT_ASYNCIFY_IMPORTS`? Currently its just `DEFAULT_ASYNCIFY_IMPORTS = ['__asyncjs__*']`.

Yeah it became smaller in my recent #20223.

> Can we instead tag all the EM_JS function that start with __asyncjs__ as .isAsync = true rather then rely on this regex?

Hm I like that idea but TBH I don't know / haven't looked into how to achieve that yet due to how they're generated.

Also note that we would still be stuck with `invoke_*` functions for dynamic dispatch, unless all of those are somehow statically marked too.

If you want to get rid of `DEFAULT_ASYNCIFY_IMPORTS` and we're not planning to add more functions there in the future, I guess we could inline it where it's used, like `invoke_*` is inlined. Seems a bit out of scope of this PR but I don't mind. https://github.com/emscripten-core/emscripten/blob/594d27d6590b381cd5844b7b3ae4af472fe5c38d/emcc.py#L2866

Reply:
Agree that its own of scope for this PR.

Reply:
> But yeah we could remove this whole combined list to `ALL_ASYNCIFY_IMPORTS` if that's preferable.

Ok I guess then the only question is this - would you prefer me to change code to use `ALL_ASYNCIFY_IMPORTS` instead?

Reply:
I think we can do that in followups if we want to.

Reply:
Sgtm, thanks for the review!